NPE during Aeon Smart Switch 6 inclusion

Greetings,
while the inclusion of this device works ok, I noticed this exception in the logs:
```
[...]
2016-09-20 12:57:31,634 | DEBUG | ZWaveInputThread | ApplicationCommandMessageClass   | 238 - org.openhab.binding.zwave - 2.0.0.201609182202 | NODE 3: Application Command Request (ALIVE:STATIC_VALUES)
2016-09-20 12:57:31,636 | DEBUG | ZWaveInputThread | ApplicationCommandMessageClass   | 238 - org.openhab.binding.zwave - 2.0.0.201609182202 | NODE 3: Incoming command class COLOR
2016-09-20 12:57:31,638 | DEBUG | ZWaveInputThread | ZWaveColorCommandClass           | 238 - org.openhab.binding.zwave - 2.0.0.201609182202 | NODE 3: Received COMMAND_CLASS_SWITCH_COLOR V1
2016-09-20 12:57:31,639 | DEBUG | ZWaveInputThread | ZWaveColorCommandClass           | 238 - org.openhab.binding.zwave - 2.0.0.201609182202 | NODE 3: Process SWITCH_COLOR_SUPPORTED_REPORT
2016-09-20 12:57:31,641 | DEBUG | ZWaveInputThread | ZWaveColorCommandClass           | 238 - org.openhab.binding.zwave - 2.0.0.201609182202 | NODE 3: Color Supported = Red(2)
2016-09-20 12:57:31,643 | DEBUG | ZWaveInputThread | ZWaveColorCommandClass           | 238 - org.openhab.binding.zwave - 2.0.0.201609182202 | NODE 3: Color Supported = Green(3)
2016-09-20 12:57:31,644 | DEBUG | ZWaveInputThread | ZWaveColorCommandClass           | 238 - org.openhab.binding.zwave - 2.0.0.201609182202 | NODE 3: Color Supported = Blue(4)
2016-09-20 12:57:31,647 | ERROR | ZWaveInputThread | ZWaveController                  | 238 - org.openhab.binding.zwave - 2.0.0.201609182202 | Exception during ZWave thread: Input 2. {}
java.lang.NullPointerException
        at org.openhab.binding.zwave.internal.protocol.commandclass.ZWaveColorCommandClass.processColorSupportedReport(ZWaveColorCommandClass.java:148)[238:org.openhab.binding.zwave:2.0.0.201609182202]
        at org.openhab.binding.zwave.internal.protocol.commandclass.ZWaveColorCommandClass.handleApplicationCommandRequest(ZWaveColorCommandClass.java:96)[238:org.openhab.binding.zwave:2.0.0.201609182202]
        at org.openhab.binding.zwave.internal.protocol.serialmessage.ApplicationCommandMessageClass.handleRequest(ApplicationCommandMessageClass.java:119)[238:org.openhab.binding.zwave:2.0.0.201609182202]
        at org.openhab.binding.zwave.internal.protocol.ZWaveController.handleIncomingRequestMessage(ZWaveController.java:247)[238:org.openhab.binding.zwave:2.0.0.201609182202]
        at org.openhab.binding.zwave.internal.protocol.ZWaveController.handleIncomingMessage(ZWaveController.java:213)[238:org.openhab.binding.zwave:2.0.0.201609182202]
        at org.openhab.binding.zwave.internal.protocol.ZWaveController.access$7(ZWaveController.java:207)[238:org.openhab.binding.zwave:2.0.0.201609182202]
        at org.openhab.binding.zwave.internal.protocol.ZWaveController$ZWaveInputThread.run(ZWaveController.java:1303)[238:org.openhab.binding.zwave:2.0.0.201609182202]
2016-09-20 12:57:35,238 | DEBUG | ESH-discovery-15 | ZWaveController                  | 238 - org.openhab.binding.zwave - 2.0.0.201609182202 | Stopping inclusion timer.
2016-09-20 12:57:35,240 | DEBUG | ESH-discovery-15 | AddNodeMessageClass              | 238 - org.openhab.binding.zwave - 2.0.0.201609182202 | Ending INCLUSION mode.
[...]
```
Here is the complete inclusion log: [aeon_smartswitch6.log.zip](https://github.com/openhab/org.openhab.binding.zwave/files/482602/aeon_smartswitch6.log.zip)

On a side note, from this log it can be seen that the binding sets the association groups for the device to `node_1_0` (that is supposed to be the controller), then `node_201_0` and `node_199_0`. I never noticed them in previous versions of the binding: is this normal?

I just tried the latest SNAPSHOT build ([build #113 on cloudbees](https://openhab.ci.cloudbees.com/job/PR-org.openhab.binding.zwave/113/)) but the exception is still there:
```
2016-09-21 10:41:59.661 [DEBUG] [ve.internal.protocol.ZWaveController] - Message: class=SendData[0x13], type=Request[0x00], priority=High, dest=255, callback=0, payload=57 00 00 02 00 D0 7F 7F 7F 7F 00 00 03 00 00 00 00 03 01 00 00 
2016-09-21 10:41:59.663 [DEBUG] [l.serialmessage.SendDataMessageClass] - NODE 8: SendData Request. CallBack ID = 87, Status = Transmission complete and ACK received(0)
2016-09-21 10:41:59.665 [DEBUG] [.serialmessage.ZWaveCommandProcessor] - Checking transaction complete: Sent Message: class=SendData[0x13], type=Request[0x00], priority=Get, dest=8, callback=87, payload=08 02 33 01 
2016-09-21 10:41:59.667 [DEBUG] [.serialmessage.ZWaveCommandProcessor] - Checking transaction complete: Recv Message: class=SendData[0x13], type=Request[0x00], priority=High, dest=255, callback=0, payload=57 00 00 02 00 D0 7F 7F 7F 7F 00 00 03 00 00 00 00 03 01 00 00 
2016-09-21 10:41:59.669 [DEBUG] [.serialmessage.ZWaveCommandProcessor] - Checking transaction complete: class=SendData, callback id=87, expected=ApplicationCommandHandler, cancelled=false      MISMATCH
2016-09-21 10:41:59.670 [DEBUG] [ve.internal.protocol.ZWaveController] - Receive queue TAKE: Length=0
2016-09-21 10:41:59.672 [DEBUG] [wave.internal.protocol.SerialMessage] - Assembled message buffer = 01 0B 00 04 00 08 04 33 02 1C 00 D0 01 
2016-09-21 10:41:59.674 [DEBUG] [ve.internal.protocol.ZWaveController] - Process Message = 01 0B 00 04 00 08 04 33 02 1C 00 D0 01 
2016-09-21 10:41:59.676 [DEBUG] [ve.internal.protocol.ZWaveController] - Message: class=ApplicationCommandHandler[0x04], type=Request[0x00], priority=High, dest=255, callback=0, payload=00 08 04 33 02 1C 00 D0 
2016-09-21 10:41:59.678 [DEBUG] [ssage.ApplicationCommandMessageClass] - NODE 8: Application Command Request (ALIVE:STATIC_VALUES)
2016-09-21 10:41:59.680 [DEBUG] [ssage.ApplicationCommandMessageClass] - NODE 8: Incoming command class COLOR
2016-09-21 10:41:59.681 [DEBUG] [.commandclass.ZWaveColorCommandClass] - NODE 8: Received COMMAND_CLASS_SWITCH_COLOR V1
2016-09-21 10:41:59.683 [DEBUG] [.commandclass.ZWaveColorCommandClass] - NODE 8: Process SWITCH_COLOR_SUPPORTED_REPORT
2016-09-21 10:41:59.684 [DEBUG] [.commandclass.ZWaveColorCommandClass] - NODE 8: Color Supported = Red(2)
2016-09-21 10:41:59.686 [DEBUG] [.commandclass.ZWaveColorCommandClass] - NODE 8: Color Supported = Green(3)
2016-09-21 10:41:59.687 [DEBUG] [.commandclass.ZWaveColorCommandClass] - NODE 8: Color Supported = Blue(4)
2016-09-21 10:41:59.689 [ERROR] [ve.internal.protocol.ZWaveController] - Exception during ZWave thread: Input 2. {}
java.lang.NullPointerException
	at org.openhab.binding.zwave.internal.protocol.commandclass.ZWaveColorCommandClass.processColorSupportedReport(ZWaveColorCommandClass.java:148)[246:org.openhab.binding.zwave:2.0.0.201609202148]
	at org.openhab.binding.zwave.internal.protocol.commandclass.ZWaveColorCommandClass.handleApplicationCommandRequest(ZWaveColorCommandClass.java:96)[246:org.openhab.binding.zwave:2.0.0.201609202148]
	at org.openhab.binding.zwave.internal.protocol.serialmessage.ApplicationCommandMessageClass.handleRequest(ApplicationCommandMessageClass.java:119)[246:org.openhab.binding.zwave:2.0.0.201609202148]
	at org.openhab.binding.zwave.internal.protocol.ZWaveController.handleIncomingRequestMessage(ZWaveController.java:247)[246:org.openhab.binding.zwave:2.0.0.201609202148]
	at org.openhab.binding.zwave.internal.protocol.ZWaveController.handleIncomingMessage(ZWaveController.java:213)[246:org.openhab.binding.zwave:2.0.0.201609202148]
	at org.openhab.binding.zwave.internal.protocol.ZWaveController.access$7(ZWaveController.java:207)[246:org.openhab.binding.zwave:2.0.0.201609202148]
	at org.openhab.binding.zwave.internal.protocol.ZWaveController$ZWaveInputThread.run(ZWaveController.java:1303)[246:org.openhab.binding.zwave:2.0.0.201609202148]
2016-09-21 10:42:04.616 [ERROR] [ocol.ZWaveController$ZWaveSendThread] - NODE 8: Timeout while sending message. Requeueing - 2 attempts left!
```
Here's the complete (hopefully in the right format) log: [aeon_smartswitch6_2.log.zip](https://github.com/openhab/org.openhab.binding.zwave/files/484763/aeon_smartswitch6_2.log.zip)

--- a/ESH-INF/thing/aeon_zw096_0_0.xml
+++ b/ESH-INF/thing/aeon_zw096_0_0.xml
@@ -52,7 +52,7 @@ Smart Switch 6<br /><h1>Overview</h1><p>Aeon Labs Smart Switch is a Z-Wave power
       <channel id="color_color" typeId="color_color">
         <label>Color</label>
         <properties>
-          <property name="binding:*:HSBType">COLOR;</property>
+          <property name="binding:*:HSBType">COLOR;colorMode=RGB</property>
           <property name="binding:*:PercentType">SWITCH_MULTILEVEL,BASIC</property>
           <property name="binding:Command:OnOffType">SWITCH_MULTILEVEL</property>
         </properties>

--- a/ESH-INF/thing/fibaro_fgrgbw_0_0.xml
+++ b/ESH-INF/thing/fibaro_fgrgbw_0_0.xml
@@ -14,8 +14,8 @@
       <channel id="switch_dimmer" typeId="switch_dimmer">
         <label>Dimmer</label>
         <properties>
-          <property name="binding:*:PercentType">SWITCH_MULTILEVEL</property>
-          <property name="binding:Command:OnOffType">SWITCH_MULTILEVEL</property>
+          <property name="binding:*:PercentType">SWITCH_MULTILEVEL,BASIC</property>
+          <property name="binding:Command:OnOffType">SWITCH_MULTILEVEL,BASIC</property>
         </properties>
       </channel>
       <channel id="sensor_power" typeId="sensor_power">
@@ -36,42 +36,43 @@
           <property name="binding:*:DecimalType">METER;type=E_KWh</property>
         </properties>
       </channel>
+      <channel id="color_color" typeId="color_color">
+        <label>Color</label>
+        <properties>
+          <property name="binding:*:HSBType">COLOR;colorMode=RGB</property>
+          <property name="binding:*:PercentType">SWITCH_MULTILEVEL,BASIC</property>
+          <property name="binding:Command:OnOffType">SWITCH_MULTILEVEL</property>
+        </properties>
+      </channel>
       <channel id="thermostat_setpoint_heating" typeId="thermostat_setpoint">
         <label>Setpoint (heating)</label>
         <properties>
           <property name="binding:*:DecimalType">THERMOSTAT_SETPOINT;type=HEATING</property>
         </properties>
       </channel>
-      <channel id="switch_dimmer1" typeId="switch_dimmer">
-        <label>Dimmer 1</label>
-        <properties>
-          <property name="binding:*:PercentType">SWITCH_MULTILEVEL:1,BASIC:1</property>
-          <property name="binding:Command:OnOffType">SWITCH_MULTILEVEL:1,BASIC:1</property>
-        </properties>
-      </channel>
       <channel id="switch_dimmer2" typeId="switch_dimmer">
-        <label>Dimmer 2</label>
+        <label>Green</label>
         <properties>
           <property name="binding:*:PercentType">SWITCH_MULTILEVEL:2,BASIC:2</property>
           <property name="binding:Command:OnOffType">SWITCH_MULTILEVEL:2,BASIC:2</property>
         </properties>
       </channel>
       <channel id="switch_dimmer3" typeId="switch_dimmer">
-        <label>Dimmer 3</label>
+        <label>Red</label>
         <properties>
           <property name="binding:*:PercentType">SWITCH_MULTILEVEL:3,BASIC:3</property>
           <property name="binding:Command:OnOffType">SWITCH_MULTILEVEL:3,BASIC:3</property>
         </properties>
       </channel>
       <channel id="switch_dimmer4" typeId="switch_dimmer">
-        <label>Dimmer 4</label>
+        <label>Blue</label>
         <properties>
           <property name="binding:*:PercentType">SWITCH_MULTILEVEL:4,BASIC:4</property>
           <property name="binding:Command:OnOffType">SWITCH_MULTILEVEL:4,BASIC:4</property>
         </properties>
       </channel>
       <channel id="switch_dimmer5" typeId="switch_dimmer">
-        <label>Dimmer 5</label>
+        <label>White</label>
         <properties>
           <property name="binding:*:PercentType">SWITCH_MULTILEVEL:5,BASIC:5</property>
           <property name="binding:Command:OnOffType">SWITCH_MULTILEVEL:5,BASIC:5</property>
@@ -282,10 +283,8 @@ Defines the required change in power to generate an unsolicited power report.<br
 
       <parameter name="config_72_1" type="integer" groupName="configuration"
                  min="1" max="10">
-        <label>72: Starting animation programm</label>
-        <description><![CDATA[
-Starting animation programm.<br /> Starting predefined program when device set to work in RGB/RGBW mode (parameter 14) - relevant for main controllers other than Home Center 2 only.<br /> Default setting: 1<br />
-        ]]></description>
+        <label>72: Starting animation program</label>
+        <description>Starting animation programm. Starting predefined program when device set to work in RGB/RGBW mode (parameter 14) - relevant for main controllers other than Home Center 2 only. Default setting: 1</description>
         <default>1</default>
       </parameter>
 

--- a/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveColorCommandClass.java
+++ b/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveColorCommandClass.java
@@ -144,18 +144,7 @@ public class ZWaveColorCommandClass extends ZWaveCommandClass implements ZWaveCo
 
         initialiseDone = true;
 
-        if (getNode() == null) {
-            logger.debug("Node is null in processColorSupportedReport");
-            return;
-        }
-
-        if (getEndpoint() == null) {
-            logger.debug("NODE {}: Endpoint is null in processColorSupportedReport", getNode().getNodeId());
-            return;
-        }
-
-        ZWaveCommandClassValueEvent zEvent = new ZWaveColorValueEvent(getNode().getNodeId(),
-                getEndpoint().getEndpointId(), colorMap);
+        ZWaveCommandClassValueEvent zEvent = new ZWaveColorValueEvent(getNode().getNodeId(), endpoint, colorMap);
         getController().notifyEventListeners(zEvent);
     }
 
@@ -188,8 +177,7 @@ public class ZWaveColorCommandClass extends ZWaveCommandClass implements ZWaveCo
             // Yes - notify of a new color
 
             logger.info("NODE {}: Color report finished {}", getNode().getNodeId(), colorMap);
-            ZWaveCommandClassValueEvent zEvent = new ZWaveColorValueEvent(getNode().getNodeId(),
-                    getEndpoint().getEndpointId(), colorMap);
+            ZWaveCommandClassValueEvent zEvent = new ZWaveColorValueEvent(getNode().getNodeId(), endpoint, colorMap);
             getController().notifyEventListeners(zEvent);
         }
     }

